components/batch-system/src/router.rs

```diff
@@ -272,14 +299,45 @@ where
         info!("[region {}] shutdown mailbox", addr);
         unsafe { &mut *self.caches.as_ptr() }.remove(&addr);
         let mut mailboxes = self.normals.lock().unwrap();
-        if let Some(mb) = mailboxes.remove(&addr) {
+        if let Some(mb) = mailboxes.map.remove(&addr) {
             mb.close();
         }
+        mailboxes
+            .alive_cnt
+            .store(mailboxes.map.len(), Ordering::Relaxed);
     }
 
     pub fn clear_cache(&self) {
         unsafe { &mut *self.caches.as_ptr() }.clear();
     }
+
+    pub fn state_cnt(&self) -> &Arc<AtomicUsize> {
+        &self.state_cnt
+    }
+
+    pub fn alive_cnt(&self) -> Arc<AtomicUsize> {
+        self.normals.lock().unwrap().alive_cnt.clone()
+    }
+
+    pub fn trace(&self) -> RouterTrace {
+        let alive = self.normals.lock().unwrap().alive_cnt.clone();
+        let total = self.state_cnt.load(Ordering::Relaxed);
+        let alive = alive.load(Ordering::Relaxed);
+        let leak = if total > alive + 1 {
```

Comment:
Do we need an assertion to make sure the total won't be less than alive?

Reply:
Should not because we get those two numbers asynchronously.